CMakeLists.txt

```diff
@@ -405,9 +405,11 @@ if(GINKGO_BUILD_MPI)
             )
             set(GINKGO_HAVE_OPENMPI_PRE_4_1_X ON)
         endif()
-        unset(valid_openmpi_version)
+        unset(gko_valid_openmpi_version)
+        unset(gko_valid_openmpi_version CACHE)
     endif()
-    unset(uses_openmpi)
+    unset(gko_uses_openmpi)
+    unset(gko_uses_openmpi CACHE)
```

Comment:
The purpose of the CACHE variables is to avoid rerunning the check every time - by unsetting them, do we increase configure time?

Reply:
I didn't think too much here but just ensure the unset correctly.
`try_compile` is not like `find_package` because they always test and reset the value to result.
to avoid checking every time, we need to guard it by ourselves.
I guess it was to adapt the changes of mpi compiler after first configuration, but I agree with changing compiler in configuration is not good practice.